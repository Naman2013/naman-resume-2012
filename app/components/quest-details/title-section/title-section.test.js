import React from 'react';
import { shallow } from 'enzyme';
import QuestDetailsTitleSection from './title-section';

const setup = (customProps = {}) => {
  const defaultProps = { preTitle: 'Test', title: 'Test', iconURL: 'foo.com' };
  const props = Object.assign({}, defaultProps, customProps);
  const shallowWrapper = shallow(<QuestDetailsTitleSection {...props} />);

  return {
    shallowWrapper,
  };
};

describe('QuestDetailsTitleSection', () => {
  const defaultProps = {
    iconURL: 'foo.com',
    preTitle: 'A guide to',
    title: 'The solar system',
  };

  it('should render correctly', () => {
    expect(setup(defaultProps).shallowWrapper).toMatchSnapshot();
  });
});
