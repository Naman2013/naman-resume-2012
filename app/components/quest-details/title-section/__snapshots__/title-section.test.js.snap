// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestDetailsTitleSection should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestDetailsTitleSection
    iconURL="foo.com"
    preTitle="A guide to"
    title="The solar system"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CenterColumn
          breakpoints={
            Array [
              "screen and (min-width: 768px)",
              "screen and (min-width: 1024px)",
              "screen and (min-width: 1366px)",
            ]
          }
          theme={
            Object {
              "background": "url(https://vega.slooh.com/assets/v4/common/White_Paper_Subtle.png)",
              "boxShadow": "rgb(191, 191, 191) 0px 11px 20px -10px",
              "padding": "70px 0",
              "position": "relative",
              "zIndex": "10",
            }
          }
          widths={
            Array [
              "620px",
              "780px",
              "940px",
            ]
          }
        >
          <div
            className="jsx-3107891613 shield-container"
          >
            <div
              className="jsx-3107891613 blue-shield"
            />
            <div
              className="jsx-3107891613 icon-container"
            >
              <img
                alt=""
                className="jsx-3107891613 icon-content"
                height="40"
                src="foo.com"
                width="40"
              />
            </div>
          </div>
          <h2
            className="jsx-3107891613 title"
          >
            <span
              className="jsx-3107891613 pre-title"
            >
              A guide to
            </span>
            The solar system
          </h2>
          <div
            className="jsx-3107891613 action-container"
          />
        </CenterColumn>,
        <JSXStyle
          css={
            Array [
              ".root.jsx-3107891613{position:relative;text-align:center;font-family:brandon-grotesque, sans-serif;color:#415671;background-color:#f2f2f2;padding-top:70px;}",
              ".title.jsx-3107891613{padding-top:80px;padding-bottom:60px;font-weight:300;text-transform:capitalize;font-size:22px;font-family:adobe-garamond-pro, serif;}",
              ".pre-title.jsx-3107891613{display:block;text-transform:uppercase;font-weight:bold;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;font-size:10px;margin-bottom:20px;font-family:brandon-grotesque, sans-serif;}",
              ".shield-container.jsx-3107891613{position:absolute;width:100%;text-align:center;height:100px;top:-30px;}",
              ".blue-shield.jsx-3107891613{position:absolute;background:url(https://vega.slooh.com/assets/v4/common/quest_shield.png);background-size:cover;background-repeat:no-repeat;height:78px;width:78px;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}",
              ".icon-content.jsx-3107891613{z-index:999;}",
              ".icon-container.jsx-3107891613{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;height:75px;}",
              ".action-container.jsx-3107891613{width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
              "@media screen and (min-width: 768px){.root.jsx-3107891613{padding-top:120px;}.title.jsx-3107891613{font-size:36px;}}",
              "@media screen and (min-width: 1024px){.title.jsx-3107891613{font-size:44px;padding-bottom:50px;}.pre-title.jsx-3107891613{margin-bottom:50px;}}",
              "@media screen and (min-width: 1366px){.root.jsx-3107891613{padding-top:160px;}.title.jsx-3107891613{font-size:42px;}}",
            ]
          }
          styleId="3107891613"
        />,
      ],
      "className": "jsx-3107891613 root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "breakpoints": Array [
            "screen and (min-width: 768px)",
            "screen and (min-width: 1024px)",
            "screen and (min-width: 1366px)",
          ],
          "children": Array [
            <div
              className="jsx-3107891613 shield-container"
            >
              <div
                className="jsx-3107891613 blue-shield"
              />
              <div
                className="jsx-3107891613 icon-container"
              >
                <img
                  alt=""
                  className="jsx-3107891613 icon-content"
                  height="40"
                  src="foo.com"
                  width="40"
                />
              </div>
            </div>,
            <h2
              className="jsx-3107891613 title"
            >
              <span
                className="jsx-3107891613 pre-title"
              >
                A guide to
              </span>
              The solar system
            </h2>,
            <div
              className="jsx-3107891613 action-container"
            />,
          ],
          "theme": Object {
            "background": "url(https://vega.slooh.com/assets/v4/common/White_Paper_Subtle.png)",
            "boxShadow": "rgb(191, 191, 191) 0px 11px 20px -10px",
            "padding": "70px 0",
            "position": "relative",
            "zIndex": "10",
          },
          "widths": Array [
            "620px",
            "780px",
            "940px",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="jsx-3107891613 blue-shield"
                />,
                <div
                  className="jsx-3107891613 icon-container"
                >
                  <img
                    alt=""
                    className="jsx-3107891613 icon-content"
                    height="40"
                    src="foo.com"
                    width="40"
                  />
                </div>,
              ],
              "className": "jsx-3107891613 shield-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "jsx-3107891613 blue-shield",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <img
                    alt=""
                    className="jsx-3107891613 icon-content"
                    height="40"
                    src="foo.com"
                    width="40"
                  />,
                  "className": "jsx-3107891613 icon-container",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "jsx-3107891613 icon-content",
                    "height": "40",
                    "src": "foo.com",
                    "width": "40",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="jsx-3107891613 pre-title"
                >
                  A guide to
                </span>,
                "The solar system",
              ],
              "className": "jsx-3107891613 title",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A guide to",
                  "className": "jsx-3107891613 pre-title",
                },
                "ref": null,
                "rendered": "A guide to",
                "type": "span",
              },
              "The solar system",
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": null,
              "className": "jsx-3107891613 action-container",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".root.jsx-3107891613{position:relative;text-align:center;font-family:brandon-grotesque, sans-serif;color:#415671;background-color:#f2f2f2;padding-top:70px;}",
            ".title.jsx-3107891613{padding-top:80px;padding-bottom:60px;font-weight:300;text-transform:capitalize;font-size:22px;font-family:adobe-garamond-pro, serif;}",
            ".pre-title.jsx-3107891613{display:block;text-transform:uppercase;font-weight:bold;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;font-size:10px;margin-bottom:20px;font-family:brandon-grotesque, sans-serif;}",
            ".shield-container.jsx-3107891613{position:absolute;width:100%;text-align:center;height:100px;top:-30px;}",
            ".blue-shield.jsx-3107891613{position:absolute;background:url(https://vega.slooh.com/assets/v4/common/quest_shield.png);background-size:cover;background-repeat:no-repeat;height:78px;width:78px;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}",
            ".icon-content.jsx-3107891613{z-index:999;}",
            ".icon-container.jsx-3107891613{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;height:75px;}",
            ".action-container.jsx-3107891613{width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
            "@media screen and (min-width: 768px){.root.jsx-3107891613{padding-top:120px;}.title.jsx-3107891613{font-size:36px;}}",
            "@media screen and (min-width: 1024px){.title.jsx-3107891613{font-size:44px;padding-bottom:50px;}.pre-title.jsx-3107891613{margin-bottom:50px;}}",
            "@media screen and (min-width: 1366px){.root.jsx-3107891613{padding-top:160px;}.title.jsx-3107891613{font-size:42px;}}",
          ],
          "styleId": "3107891613",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CenterColumn
            breakpoints={
              Array [
                "screen and (min-width: 768px)",
                "screen and (min-width: 1024px)",
                "screen and (min-width: 1366px)",
              ]
            }
            theme={
              Object {
                "background": "url(https://vega.slooh.com/assets/v4/common/White_Paper_Subtle.png)",
                "boxShadow": "rgb(191, 191, 191) 0px 11px 20px -10px",
                "padding": "70px 0",
                "position": "relative",
                "zIndex": "10",
              }
            }
            widths={
              Array [
                "620px",
                "780px",
                "940px",
              ]
            }
          >
            <div
              className="jsx-3107891613 shield-container"
            >
              <div
                className="jsx-3107891613 blue-shield"
              />
              <div
                className="jsx-3107891613 icon-container"
              >
                <img
                  alt=""
                  className="jsx-3107891613 icon-content"
                  height="40"
                  src="foo.com"
                  width="40"
                />
              </div>
            </div>
            <h2
              className="jsx-3107891613 title"
            >
              <span
                className="jsx-3107891613 pre-title"
              >
                A guide to
              </span>
              The solar system
            </h2>
            <div
              className="jsx-3107891613 action-container"
            />
          </CenterColumn>,
          <JSXStyle
            css={
              Array [
                ".root.jsx-3107891613{position:relative;text-align:center;font-family:brandon-grotesque, sans-serif;color:#415671;background-color:#f2f2f2;padding-top:70px;}",
                ".title.jsx-3107891613{padding-top:80px;padding-bottom:60px;font-weight:300;text-transform:capitalize;font-size:22px;font-family:adobe-garamond-pro, serif;}",
                ".pre-title.jsx-3107891613{display:block;text-transform:uppercase;font-weight:bold;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;font-size:10px;margin-bottom:20px;font-family:brandon-grotesque, sans-serif;}",
                ".shield-container.jsx-3107891613{position:absolute;width:100%;text-align:center;height:100px;top:-30px;}",
                ".blue-shield.jsx-3107891613{position:absolute;background:url(https://vega.slooh.com/assets/v4/common/quest_shield.png);background-size:cover;background-repeat:no-repeat;height:78px;width:78px;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}",
                ".icon-content.jsx-3107891613{z-index:999;}",
                ".icon-container.jsx-3107891613{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;height:75px;}",
                ".action-container.jsx-3107891613{width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
                "@media screen and (min-width: 768px){.root.jsx-3107891613{padding-top:120px;}.title.jsx-3107891613{font-size:36px;}}",
                "@media screen and (min-width: 1024px){.title.jsx-3107891613{font-size:44px;padding-bottom:50px;}.pre-title.jsx-3107891613{margin-bottom:50px;}}",
                "@media screen and (min-width: 1366px){.root.jsx-3107891613{padding-top:160px;}.title.jsx-3107891613{font-size:42px;}}",
              ]
            }
            styleId="3107891613"
          />,
        ],
        "className": "jsx-3107891613 root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "breakpoints": Array [
              "screen and (min-width: 768px)",
              "screen and (min-width: 1024px)",
              "screen and (min-width: 1366px)",
            ],
            "children": Array [
              <div
                className="jsx-3107891613 shield-container"
              >
                <div
                  className="jsx-3107891613 blue-shield"
                />
                <div
                  className="jsx-3107891613 icon-container"
                >
                  <img
                    alt=""
                    className="jsx-3107891613 icon-content"
                    height="40"
                    src="foo.com"
                    width="40"
                  />
                </div>
              </div>,
              <h2
                className="jsx-3107891613 title"
              >
                <span
                  className="jsx-3107891613 pre-title"
                >
                  A guide to
                </span>
                The solar system
              </h2>,
              <div
                className="jsx-3107891613 action-container"
              />,
            ],
            "theme": Object {
              "background": "url(https://vega.slooh.com/assets/v4/common/White_Paper_Subtle.png)",
              "boxShadow": "rgb(191, 191, 191) 0px 11px 20px -10px",
              "padding": "70px 0",
              "position": "relative",
              "zIndex": "10",
            },
            "widths": Array [
              "620px",
              "780px",
              "940px",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="jsx-3107891613 blue-shield"
                  />,
                  <div
                    className="jsx-3107891613 icon-container"
                  >
                    <img
                      alt=""
                      className="jsx-3107891613 icon-content"
                      height="40"
                      src="foo.com"
                      width="40"
                    />
                  </div>,
                ],
                "className": "jsx-3107891613 shield-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "jsx-3107891613 blue-shield",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <img
                      alt=""
                      className="jsx-3107891613 icon-content"
                      height="40"
                      src="foo.com"
                      width="40"
                    />,
                    "className": "jsx-3107891613 icon-container",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "alt": "",
                      "className": "jsx-3107891613 icon-content",
                      "height": "40",
                      "src": "foo.com",
                      "width": "40",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "img",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="jsx-3107891613 pre-title"
                  >
                    A guide to
                  </span>,
                  "The solar system",
                ],
                "className": "jsx-3107891613 title",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A guide to",
                    "className": "jsx-3107891613 pre-title",
                  },
                  "ref": null,
                  "rendered": "A guide to",
                  "type": "span",
                },
                "The solar system",
              ],
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": null,
                "className": "jsx-3107891613 action-container",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".root.jsx-3107891613{position:relative;text-align:center;font-family:brandon-grotesque, sans-serif;color:#415671;background-color:#f2f2f2;padding-top:70px;}",
              ".title.jsx-3107891613{padding-top:80px;padding-bottom:60px;font-weight:300;text-transform:capitalize;font-size:22px;font-family:adobe-garamond-pro, serif;}",
              ".pre-title.jsx-3107891613{display:block;text-transform:uppercase;font-weight:bold;-webkit-letter-spacing:2px;-moz-letter-spacing:2px;-ms-letter-spacing:2px;letter-spacing:2px;font-size:10px;margin-bottom:20px;font-family:brandon-grotesque, sans-serif;}",
              ".shield-container.jsx-3107891613{position:absolute;width:100%;text-align:center;height:100px;top:-30px;}",
              ".blue-shield.jsx-3107891613{position:absolute;background:url(https://vega.slooh.com/assets/v4/common/quest_shield.png);background-size:cover;background-repeat:no-repeat;height:78px;width:78px;left:50%;-webkit-transform:translateX(-50%);-ms-transform:translateX(-50%);transform:translateX(-50%);}",
              ".icon-content.jsx-3107891613{z-index:999;}",
              ".icon-container.jsx-3107891613{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;height:75px;}",
              ".action-container.jsx-3107891613{width:100%;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-webkit-justify-content:center;-ms-flex-pack:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
              "@media screen and (min-width: 768px){.root.jsx-3107891613{padding-top:120px;}.title.jsx-3107891613{font-size:36px;}}",
              "@media screen and (min-width: 1024px){.title.jsx-3107891613{font-size:44px;padding-bottom:50px;}.pre-title.jsx-3107891613{margin-bottom:50px;}}",
              "@media screen and (min-width: 1366px){.root.jsx-3107891613{padding-top:160px;}.title.jsx-3107891613{font-size:42px;}}",
            ],
            "styleId": "3107891613",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
